ubcpi/ubcpi.py

```diff
@@ -0,0 +1,468 @@
+"""A Peer Instruction tool for edX by the University of British Columbia."""
+import os
+import random
+from copy import deepcopy
+import uuid
+
+from django.core.exceptions import PermissionDenied
+
+import pkg_resources
+from webob import Response
+
+from xblock.core import XBlock
+from xblock.exceptions import JsonHandlerError
+from xblock.fields import Scope, String, List, Dict, Integer, DateTime, Float
+from xblock.fragment import Fragment
+from answer_pool import offer_answer, validate_seeded_answers, get_other_answers
+import persistence as sas_api
+from serialize import parse_from_xml
+
+STATUS_NEW = 0
+STATUS_ANSWERED = 1
+STATUS_REVISED = 2
+
+# this is a limitation in the database submissions_submission.raw_answer field, which is a TEXT and has limit
+# of 64k in size. Because we are storing rationale and revised rationale both in the the field, the max size
+# for the rationale is half
+MAX_RATIONALE_SIZE = 32000
+
+
+# noinspection all
+class MissingDataFetcherMixin:
+    """ Copied from https://github.com/edx/edx-ora2/blob/master/openassessment/xblock/openassessmentblock.py """
```

Comment:
As far as our tests go, we didn't see any problem for that piece of code. We learnt how to use submission API by copying that piece(there isn't a lot of docs on how to generate that student_item_dict for submission API). As we are pretty new to edx and xblock development, If possible, could you ask someone who is more familiar to check it?